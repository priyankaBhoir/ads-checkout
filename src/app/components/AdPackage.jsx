import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class AdPackage extends Component {
	static propTypes = {
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    description: PropTypes.array.isRequired
	}
	constructor(props) {
		super(props);
	}

	render() {
		return (
			<div className="icon-block" style={{height: 200}}>
				<h5 className="center">{this.props.title}</h5>
				<ul>{this.props.description.map((desc, index) => <li key={index}> {desc} </li>)} </ul>
				<div className="ad-price"> $ {this.props.price} </div>
			</div>
		)
	}
}
