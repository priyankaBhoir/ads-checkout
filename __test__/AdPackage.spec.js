import React from 'react'
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer'
import AdPackage from '../src/app/components/AdPackage'

// Snapshot for Home React Component
describe('Ad --- Snapshot',()=>{
    it('+++capturing Snapshot of AdPackage', () => {
        const renderedValue =  renderer.create(<AdPackage id="classic"
            title="Classic"
            price={299}
            description={["first", "second"]} />).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });

});
///*******************************************************************************************************
describe('Shallow Render REACT COMPONENTS :: Ad',()=>{
    let wrapper

    beforeEach(()=>{
        wrapper = shallow(<AdPackage id="classic"
            title="Classic"
            price={299}
            description={['first', 'second']} /> )
        
    })

    it('+++ render the DUMB component', () => {
       expect(wrapper.length).toEqual(1)
    });

    it('+++ contains header - h5', () => {
        expect(wrapper.contains(<h5 className="center">Classic</h5>)).toBe(true)
    });
    it('+++ header value is title given in props', () => {
        expect(wrapper.find('h5').get(0).props.children).toBe("Classic")
    });
    it('+++ Ad blocks displays price for ad', () => {
        expect(wrapper.find('.ad-price').at(0)
                .equals(<div className="ad-price"> $ 299 </div>))
                .toBe(true)
    });
    
});
