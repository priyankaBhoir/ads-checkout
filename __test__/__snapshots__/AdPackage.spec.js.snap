// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ad --- Snapshot +++capturing Snapshot of AdPackage 1`] = `
<div
  className="icon-block"
  style={
    Object {
      "height": 200,
    }
  }
>
  <h5
    className="center"
  >
    Classic
  </h5>
  <ul>
    <li>
       
      first
       
    </li>
    <li>
       
      second
       
    </li>
     
  </ul>
  <div
    className="ad-price"
  >
     $ 
    299
     
  </div>
</div>
`;
